lib/zendesk_api/client.rb

```diff
@@ -37,10 +37,11 @@ class Client
     def method_missing(method, *args, &block)
       method = method.to_s
       options = args.last.is_a?(Hash) ? args.pop : {}
-
+      resource_class = method_as_class(method)
+      options = options.merge(resource_class.default_collection_options) if resource_class.respond_to?(:default_collection_options)
       unless config.use_resource_cache
-        raise "Resource for #{method} does not exist" unless method_as_class(method)
-        return ZendeskAPI::Collection.new(self, method_as_class(method), options)
+        raise "Resource for #{method} does not exist" unless resource_class
+        return ZendeskAPI::Collection.new(self, resource_class, options)
```

Comment:
Method missing has a huge footprint and this requires some extra confidence, seems fine, but I might want to dig deeper.